docs/Developers/Debug_Bears.rst

```diff
@@ -122,3 +122,74 @@ Or to debug all bears specified by ``bears`` setting:
     bears = PEP8Bear,MypyBear
     files = <filename>
     debug_bears = True
+
+Inspect Bear settings
+---------------------
+
+External pdb command ``settings`` has been included to inspect the bear
+settings in the debugging environment.(``settings`` command display's the
+updated setting of a Bear)
+
+Bellow is the code of simple CharCheckBear which check whether the
+passed message contains character more than 79.
+
+.. code:: python
+
+    import logging
+
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class CharCheckBear(LocalBear):
+        def run(self, filename, file, message: str, line_length: int = 79):
+            """
+            Detect if a passed string contain characters more than 79.
+
+            :param message:
+                enter the message.
+            """
+            if len(message)<=79:
+                yield self.new_result(message=message, file=filename)
+            else:
+                yield self.new_result(message='message contains character '
+                                              'more than 79', file=filename)
+
+In the above example ``line_length`` is optional and ``message` is non optional
+setting of a CharCheckBear. After invoking debugger on CharCheckBear ``coala``
+will first ask the value of all unpassed non optional setting i.e, ``message``
+in our case and after debugger will get invoke on ``run()`` method of
+CharCheckBear, pdb ``settings`` command will be used to display the updated
+optional and non optional settings of a Bear in debugging environment i.e.,
+
+::
+
+    Please enter a value for the setting "message" (enter the message.) needed
+    by CharCheckBear for section "cli":
+    Hello World
+    [DEBUG][17:25:09] Platform Linux -- Python 3.6.5, coalib
+    0.12.0.dev99999999999999
+    Executing section cli...
+    [DEBUG][17:25:09] Files that will be checked:
+    /home/Voldemort/test/mytest.py
+    [DEBUG][17:25:09] coala is run only on changed files, bears' log messages
+    from previous runs may not appear. You may use the `--flush-cache` flag to
+    see them.
+    [DEBUG][17:25:09] Running bear CharCheckBear...
+    > /home/Voldemort/coala-bears/bears/general/CharCheckBear.py(14)run()
```

Comment:
hmmm just recognized, "Voldemort" is maybe not good path component here ;) but we can change that afterwards, since untouched parts of the file has the same paths.